$OpenBSD: patch-src_twisted_python__setup_py,v 1.2 2018/05/25 10:42:38 pvk Exp $

Add suffix for py3 FLAVOR

Don't install files used to build the extension modules it uses:
https://github.com/twisted/twisted/commit/80e67a6775a190cba77e0d4229d12b9859ba12fc

Index: src/twisted/python/_setup.py
--- src/twisted/python/_setup.py.orig
+++ src/twisted/python/_setup.py
@@ -138,15 +138,15 @@ _EXTRAS_REQUIRE['osx_platform'] = _EXTRAS_REQUIRE['mac
 
 # Scripts provided by Twisted on Python 2 and 3.
 _CONSOLE_SCRIPTS = [
-    "ckeygen = twisted.conch.scripts.ckeygen:run",
-    "cftp = twisted.conch.scripts.cftp:run",
-    "conch = twisted.conch.scripts.conch:run",
-    "mailmail = twisted.mail.scripts.mailmail:run",
-    "pyhtmlizer = twisted.scripts.htmlizer:run",
-    "tkconch = twisted.conch.scripts.tkconch:run",
-    "trial = twisted.scripts.trial:run",
-    "twist = twisted.application.twist._twist:Twist.main",
-    "twistd = twisted.scripts.twistd:run",
+    "ckeygen${MODPY_BIN_SUFFIX} = twisted.conch.scripts.ckeygen:run",
+    "cftp${MODPY_BIN_SUFFIX} = twisted.conch.scripts.cftp:run",
+    "conch${MODPY_BIN_SUFFIX} = twisted.conch.scripts.conch:run",
+    "mailmail${MODPY_BIN_SUFFIX} = twisted.mail.scripts.mailmail:run",
+    "pyhtmlizer${MODPY_BIN_SUFFIX} = twisted.scripts.htmlizer:run",
+    "tkconch${MODPY_BIN_SUFFIX} = twisted.conch.scripts.tkconch:run",
+    "trial${MODPY_BIN_SUFFIX} = twisted.scripts.trial:run",
+    "twist${MODPY_BIN_SUFFIX} = twisted.application.twist._twist:Twist.main",
+    "twistd${MODPY_BIN_SUFFIX} = twisted.scripts.twistd:run",
     ]
 
 
@@ -250,6 +250,9 @@ def getSetupArgs(extensions=_EXTENSIONS):
         },
         cmdclass=command_classes,
         include_package_data=True,
+	exclude_package_data={
+	    "": ["*.c", "*.h", "*.pxi", "*.pyx", "build.bat"],
+	},
         zip_safe=False,
         extras_require=_EXTRAS_REQUIRE,
         package_dir={"": "src"},
